src/Microsoft.TemplateEngine.Core/Util/ProcessorState.cs

```diff
@@ -72,12 +74,15 @@ public ProcessorState(Stream source, Stream target, int bufferSize, int flushThr
             CurrentBuffer = new byte[bufferSize];
             CurrentBufferLength = source.Read(CurrentBuffer, 0, CurrentBuffer.Length);
 
-            byte[] bom;
             Encoding encoding = EncodingUtil.Detect(CurrentBuffer, CurrentBufferLength, out bom);
             Encoding = encoding;
             CurrentBufferPosition = bom.Length;
             CurrentSequenceNumber = bom.Length;
-            target.Write(bom, 0, bom.Length);
+            if(bom.Length > 0)
```

Comment:
if it's possible bom is null, is the pattern in the class to throw?  If not, we should test for null & adapt the class' error handling pattern.

Reply:
the bom is instantiated in the EncodingUtil.Detect. Exactly as it was before.

Reply:
I think it makes sense to continue with the error handling method that was used in the class. I also wish to have a discussion about this with you to better evaluate pros and cons of each approach.

For this case, EncodingUtil.Detect doesn't seem to return a null value for bom. So I think we don't need to do any checks for now. We still need to make sure this doesn't became an issue in the future:

If we don't have nullable reference types enabled, then we should make sure that `EncodingUtil.Detect` doesn't return null in the future either. I suggest writing some tests that checks different `EncodingUtil.Detect` cases to make sure it never returns null.

If we have nullable reference types enabled in the project, having a not-nullable return type is good enough security measure for us in my opinion.